src/main/java/chess/controller/SpringController.java

```diff
@@ -25,72 +23,89 @@ public SpringController(ChessService chessService) {
-                       Model model, RedirectAttributes redirectAttributes) throws UnsupportedEncodingException {
-        String encodedParam = URLEncoder.encode(gameName, "UTF-8");
-        try {
-            String command = makeCommand(from, to);
-            chessService.move(command);
-            if (chessService.isEnd()) {
-                return "redirect:/game/" + encodedParam + "/end";
-            }
-        } catch (IllegalArgumentException e) {
-            redirectAttributes.addAttribute("error", e.getMessage());
-        }
+                       Model model) {
+        model.addAttribute("id", id);
 
-        model.addAttribute("gameName", gameName);
+        String command = makeCommand(from, to);
+        chessService.move(id, command);
 
-        return "redirect:/game/" + encodedParam;
+        if (chessService.isEnd(id)) {
+            return "redirect:/game/" + id + "/end";
+        }
+        return "redirect:/game/" + id;
     }
 
-    @GetMapping("/game/{gameName}/end")
-    public String end(@PathVariable String gameName, Model model) {
-        String winTeamName = chessService.finish(Command.from("end"));
-        List<String> chessBoard = chessService.getCurrentChessBoard();
+    @GetMapping("/game/{id}/end")
+    public String end(@PathVariable Long id, Model model) {
+        String winTeamName = chessService.finish(id, Command.from("end"));
+        List<String> chessBoard = chessService.getCurrentChessBoard(id);
 
         model.addAttribute("winTeam", winTeamName);
         model.addAttribute("chessboard", chessBoard);
-        model.addAttribute("gameName", gameName);
+        model.addAttribute("id", id);
 
         return "chess";
     }
 
-    @GetMapping("/game/{gameName}/status")
-    public String status(@PathVariable String gameName, Model model) {
-        Map<Team, Double> score = chessService.getScore();
-        List<String> chessBoard = chessService.getCurrentChessBoard();
+    @GetMapping("/game/{id}/status")
+    public String status(@PathVariable Long id, Model model) {
+        Map<Team, Double> score = chessService.getScore(id);
+        List<String> chessBoard = chessService.getCurrentChessBoard(id);
 
         model.addAttribute("blackScore", score.get(Team.BLACK));
         model.addAttribute("whiteScore", score.get(Team.WHITE));
         model.addAttribute("chessboard", chessBoard);
-        model.addAttribute("gameName", gameName);
+        model.addAttribute("id", id);
 
         return "chess";
     }
 
+    @PostMapping("/delete")
+    public String delete(@RequestParam(value = "game_name", required = false) String gameName,
+                         @RequestParam(value = "password", required = false) String password) {
+        State state = chessService.findStateByGameNameAndPassword(gameName, password);
+        chessService.deleteByGameNameAndPassword(state, gameName, password);
+
+        return "redirect:/";
```

Comment:
```suggestion
        chessService.deleteBy(id, password);

        return "redirect:/";
```
또 컨트롤러 내에서 비즈니스의 플로우를 작성하지마시고, 서비스에 위임하는게 좋아보입니다.
다른 컨트롤러도 적용해주세요~

Reply:
컨트롤러는 "gameName과 password를 통해서 삭제를 한다" 라는 것에만 관심이 있지, "어떻게(how)" 삭제를 하는지는 알 필요가 없다고 생각합니다. 즉, 말씀해주신대로 비즈니스 흐름(어떤 메소드를 호출하는지 그리고 어떤 순서로 호출하는지)에 대해서는 알 필요가 없고, 이를 서비스에 위임하는게 적절해 보입니다!! 수정해보도록 하겠습니다!

(이전까지는 deleteByGameNameAndPassword()와 같이 메소드명을 작성하였는데, deleteBy(gameName, password)와 같은 네이밍으로 메소드명도 짧으면서 의미를 확실히 전달할 수 있다는 생각이 드네요!! 이 부분도 한 번 반영해보도록 하겠습니다!😃)